### PR TITLE
Soho 5861 maintain format dataset bugfix

### PR DESCRIPTION
**Pull request by tjamesallen15**
> Explain the **details** for making this change. What existing problem does the pull request solve?
Fix for SOHO-5861 Issue

> **Related issue (required)**:
https://jira.infor.com/browse/SOHO-5861

> **Steps necessary to review your pull request (required)**:

> Other Details:

### CODE REVIEW

**Comment by nickwynja** [comment]
@tjamesallen15 Can you please fix these lint errors and push again? https://travis-ci.com/infor-design/enterprise/builds/74354737#L488-L494

Thanks!

**Comment by tjamesallen15** [comment]
I pushed a followup fix for the lint errors.

File: src/components/locale/locale.js [hunk 1]
```diff
@@ -93,6 +93,114 @@ const Locale = {  // eslint-disable-line
     return isThere;
   },
 
+  /**
+   * Transforms the date value into string with date format
+   * @param {string | Date} value The value that needs to be parsed.
+   * @param {string} sourceFormat The dateFormat specified for the dataset.
+   * @returns {string} string format of the date value.
+   */
+  parseDateToSource(value, sourceFormat) {
```
Comment (tmcconechy) [inline src/components/locale/locale.js]: I think we are duplicating lots of functionality here. We have two functions already that do this sort of thing.

Locale.parseDate()  // takes a string object in particular format and returns a data object
Locale.formatDate() // takes a date object and returns a particular format

See tests
https://github.com/infor-design/enterprise/blob/master/test/components/locale/locale.func-spec.js#L268

https://github.com/infor-design/enterprise/blob/master/test/components/locale/locale.func-spec.js#L416

File: src/components/locale/locale.js [hunk 2]
```diff
@@ -482,13 +482,24 @@ const Locale = {  // eslint-disable-line
     }
 
     if (dateFormat.indexOf(' ') === -1 && dateFormat.indexOf('.') === -1 && dateFormat.indexOf('/') === -1 && dateFormat.indexOf('-') === -1) {
+      // Remove delimeter for the data string.
```
[the remark this answers is not in the document]
Comment (tjamesallen15) [reply src/components/locale/locale.js]: Sure, tim! :)

File: test/components/locale/locale.func-spec.js [hunk 3]
```diff
@@ -800,4 +800,16 @@ describe('Locale API', () => {
       expect(Locale.currentLocale.name).toEqual('fi-FI'); // not found so equals last one
     }
   });
+
+  it('Should parse dates with and without spaces, dash, comma format', () => {
+    Locale.set('en_US');
```
Comment (tmcconechy) [inline test/components/locale/locale.func-spec.js]: en-US

**Comment by tmcconechy** [comment]
I pushed a small fix. But what i am wondering mainly is how to test this. Can you add the steps to test and confirm.

Thanks

**Comment by tjamesallen15** [comment]
Steps to Test:
Link: http://localhost:4000/components/datagrid/test-source-format

Source Format: yyyyMMdd
Change One Row of OrderDate

It should show the in the UI the dateFormat of yyyy-MM-dd (Ex. 2014-07-03)
However orderDate inside the dataset should be in sourceFormat (Ex. 20140703)

To check an alert dialog will show..

**Pull request merged**